### ISSUE TITLE
`std.meta.hasMethod` should not return true for multi-pointer types (or `std.fmt` should check for this)

### ISSUE THREAD

**Issue by euclidianAce** [ziglang/zig#18708]
### Zig Version

0.12.0-dev.2341+92211135f

### Steps to Reproduce and Observed Behavior

Attempt to `build-exe` the following

```zig
// example.zig

pub fn main() void {
    const arr = [_]SomeType{ .{ .x = 1 }, .{ .x = 2 }, .{ .x = 3 } };
    printSlice(&arr);
}

fn printSlice(slice: []const SomeType) void {
    std.debug.print("{any}\n", .{slice});
}

const SomeType = struct {
    x: i32,

    pub fn format(
        self: @This(),
        comptime _: []const u8,
        _: std.fmt.FormatOptions,
        writer: anytype,
    ) !void {
        try writer.print("(x={})", .{self.x});
    }
};

const std = @import("std");
```

### Expected Behavior

The previous behavior of each element of the slice being formatted, but now `std.fmt` thinks that the slice itself has a method `format` because the child type `SomeType` does.

**Comment by ianprime0509** [comment]
Related PR: #18674

**Comment by andrewrk** [comment]
Invalid bug report. It does not specify Expected Behavior, making it difficult to check if the issue has been fixed or not.

Fortunately, the one marked as a duplicate of this (#18743) does fill out the Expected Behavior field, enabling me to verify that, indeed, this has been solved by #18674, which landed in 60308550978be821515323256af17e0b8a610b1c.
